### PR TITLE
Move to CircleCI 2.0

### LINKED ISSUE #76: Migrate CI off CircleCI 1.0
CircleCI have deprecated their 1.0 config.
Perhaps migrate to CircleCI 2.0.

### PR DESCRIPTION
**Pull request by bboreham**
Fixes #76

### CODE REVIEW

**Comment by dholbach** [comment]
Looks like `virtualenv` is not installed?

**Comment by bboreham** [comment]
First I tried building with the Python image and it couldn't do the Go program, and now the Go image can't do Python, so the next step would be to try two images in two CircleCI jobs.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+jobs:
+  build-test:
+    <<: *defaults
+    steps:
+    - checkout
+    - run:
+        name: Dependencies
+        command: |
+          pip install virtualenv
```
Comment (dholbach) [inline .circleci/config.yml]:
```suggestion
          apt install python-pip
          pip install virtualenv
```

Comment (dholbach) [reply .circleci/config.yml]: Haven't tested it, but it looks like `pip` is missing (https://circleci.com/gh/weaveworks/kubediff/150) and we were using pip and python2 before: https://github.com/weaveworks/kubediff/blob/1501a2e89b8a8ac0fe3b21daecc2b6994be2703c/Dockerfile#L5

Comment (bboreham) [reply .circleci/config.yml]: It's working now; perhaps you caught it in one of the variants I tried.

**Comment by bboreham** [comment]
> the next step would be to try two images in two CircleCI jobs

I decided this was over-complicated, requiring images to be saved from one step and brought back in another, so moved everything into one 'job' and installed the necessary Python components into a Go build image.

**Pull request merged**